app/models/media.rb

```diff
@@ -80,4 +83,17 @@ def error_but_not_item_error?
   def update_item_error
     errors.add(:item, :reattach) if error_but_not_item_error?
   end
+
+  def non_image_cannot_be_featured
+    errors.add(:featured, :non_image) if kind_value != 'image' && featured
+  end
+
+  def one_featured_image_per_loan
+    featured_presently = self.class.where(media_attachable_type: 'Project', media_attachable_id: media_attachable_id).
```

Comment:
I think, as written, this will remove the featured status if you change the caption or another field on the item that's already featured. Maybe we should reverse this logic (if the current item is set to featured, use `update_all` to remove the featured flag from all the other media for that attachable type)

Reply:
this is genius!

thank you

I was originally trying to skip the loop with validations hence the validate: false option